noir-projects/noir-contracts/contracts/token_contract/src/main.nr

```diff
@@ -24,7 +24,6 @@ contract Token {
             encrypted_event_emission::encode_and_encrypt_event_unconstrained,
```

Comment:
The changes in this file are a bit unfortunate because originally we didn't have to deal with "transient" storage as we were able to just pass the values as args to the teardown function.

I decided to use the transient storage approach here as well because:
1. I forgot about this flow and I stumbled upon it only after I changed the macros 😂 (and after the change we cannot instantiate finalization payload from raw values as it expects context and loading from storage),
2. I think we might need to go with the transient storage approach because that will allow us to eventually not force user to define the note log length manually which I think is extremely important (deriving that manually essentially makes the dev a compiler).